### PR TITLE
feat: OAuth2 cli config

### PR DESCRIPTION
**Pull request by vsukhin**
## Pull request description 

OAuth2 cli config

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: cmd/kubectl-testkube/config/data.go [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+)
+
 type Data struct {
-	AnalyticsEnabled bool   `json:"analyticsEnabled,omitempty"`
-	Namespace        string `json:"namespace,omitempty"`
-	Initialized      bool   `json:"initialized,omitempty"`
+	AnalyticsEnabled bool       `json:"analyticsEnabled,omitempty"`
+	Namespace        string     `json:"namespace,omitempty"`
+	Initialized      bool       `json:"initialized,omitempty"`
+	OAuth2Data       OAuth2Data `json:"oauth2_data"`
```
Comment (exu) [inline cmd/kubectl-testkube/config/data.go:11]: Camel case in Json

File: docs/cli/kubectl-testkube_delete_webhook.md [hunk 2]
```diff
@@ -21,9 +21,11 @@ kubectl-testkube delete webhook <webhookName> [flags]
 ### Options inherited from parent commands
 
 ```
-      --analytics-enabled   enable analytics (default true)
+      --analytics-enabled   enable analytics
+  -w, --api-uri string      api uri, default value read from config if set
```
Comment (exu) [inline docs/cli/kubectl-testkube_delete_webhook.md:25]: `-w` could be a little counter intuitive

File: pkg/api/v1/client/direct_transport.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	"path/filepath"
+	"strings"
+
+	"github.com/kubeshop/testkube/pkg/executor/output"
+	phttp "github.com/kubeshop/testkube/pkg/http"
+	"github.com/kubeshop/testkube/pkg/problem"
+	"golang.org/x/oauth2"
+)
+
+// NewDirectTransport returns new proxy transport
```
Comment (exu) [inline pkg/api/v1/client/direct_transport.go:21]: shouldn't it be like `NewDirectTransport returns new direct transport`?

Comment (exu) [inline pkg/api/v1/client/direct_transport.go:21]: not sure if `Transport` is good name here - I feel transport more like  HTTP / GRPC thing this one is more like `Client`?

**Review by exu — APPROVED** [review]
Some minor comments

File: docs/cli/kubectl-testkube_delete_webhook.md (hunk 2 above)
Comment (vsukhin) [reply docs/cli/kubectl-testkube_delete_webhook.md]: a - should be better)

File: pkg/api/v1/client/direct_transport.go (hunk 3 above)
Comment (vsukhin) [reply pkg/api/v1/client/direct_transport.go]: renamed

Comment (vsukhin) [reply pkg/api/v1/client/direct_transport.go]: renamed

**Pull request merged**